jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -187,25 +187,97 @@ public ImageAndRegistryClient call()
    * @throws BadContainerConfigurationFormatException if the container configuration is in a bad
    *     format
    */
-  private Image pullBaseImage(
+  private ImagesAndRegistryClient pullBaseImages(
```

Comment:
I think it makes sense for this method to return `List<Image>`. Basically, not slipping the responsibilty of returning `ImagesAndRegistryClient` to this "helper" method, which can only have a local focus on downloading base images if viewed in isolation.